src/components/App/App.js

```diff
@@ -101,21 +152,159 @@ function App() {
       <CurrentUserContext.Provider value={currentUser}>
         <SavedArticlesContext.Provider value={savedArticles}>
           <div className='app'>
-            <PopupWithForm
-              isOpen={isPopupOpen}
-              setIsOpen={setIsPopupOpen}
-              signUpReq={mainApi.signUp.bind(mainApi)}
-              signInReq={mainApi.signIn.bind(mainApi)}
-              reportError={mainApi.reportError.bind(mainApi)}
-            />
+            <Popup isOpen={isPopupOpen} onClose={closePopup} type={popupType}>
+              <Form
```

Comment:
Please, see my advice once more https://disk.yandex.ru/i/yzUk-RsxZgItNQ . I suggested to make `Popup` to be used inside `PopupWithForm` (not separately)